### PR TITLE
opentelemetry-proto: add package_type, clean up

### PR DESCRIPTION
**Pull request by valgur**
Specify library name and version:  **opentelemetry-proto/***

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/opentelemetry-proto/all/test_package/conanfile.py [hunk 1]
```diff
@@ -1,34 +1,24 @@
...
-        # TODO: to remove in conan v2
-        if self._res_folder == "":
-            self._res_folder = self.deps_user_info["opentelemetry-proto"].proto_root
-
-        assert os.path.isfile(os.path.join(self._res_folder, "opentelemetry", "proto", "common", "v1", "common.proto"))
+        if conan_version.major >= 2:
+            res_folder = self.dependencies["opentelemetry-proto"].conf_info.get("user.opentelemetry-proto:proto_root")
+        else:
+            # TODO: to remove in conan v2
+            res_folder = self.deps_user_info["opentelemetry-proto"].proto_root
```
Comment (RubenRBS) [inline recipes/opentelemetry-proto/all/test_package/conanfile.py:21]: Can't this just made to work in both versions with the same v2 approach?

File: recipes/opentelemetry-proto/all/conanfile.py [hunk 2]
```diff
@@ -1,43 +1,42 @@
...
-    topics = ("opentelemetry", "telemetry", "otlp")
+    topics = ("opentelemetry", "telemetry", "otlp", "pre-built")
+
+    package_type = "application"
     settings = "os", "arch", "compiler", "build_type"
-    no_copy_source = True
 
     def layout(self):
-        basic_layout(self, src_folder="src")
+        pass
```
Comment (RubenRBS) [inline recipes/opentelemetry-proto/all/conanfile.py:20]: Why `pass` here? I would expected to have a layout here if it used to work (Or was it broken before?)

**Review by RubenRBS — COMMENTED** [review]
Thanks! Some comments though

File: recipes/opentelemetry-proto/all/test_package/conanfile.py (hunk 1 above)
Comment (valgur) [reply recipes/opentelemetry-proto/all/test_package/conanfile.py:21]: @uilianres used this approach for ua-nodeset: https://github.com/conan-io/conan-center-index/commit/5c80fc5d4dbbb719b6e7f743abe3925d5a626f85

File: recipes/opentelemetry-proto/all/conanfile.py (hunk 2 above)
Comment (valgur) [inline recipes/opentelemetry-proto/all/conanfile.py:20]:
```suggestion
        basic_layout(self, src_folder="src") 
```

Comment (valgur) [reply recipes/opentelemetry-proto/all/conanfile.py:20]: That was accidental. Will revert.

File: recipes/opentelemetry-proto/all/test_package/conanfile.py (hunk 1 above)
Comment (uilianries) [reply recipes/opentelemetry-proto/all/test_package/conanfile.py:21]: There is a better approach, you can store it in a file. It's well used by other recipes:


```python
def generate(self):
    save(self, self.dependencies["opentelemetry-proto"].conf_info.get("user.opentelemetry-proto:proto_root"), "proto_root")



def test(self):
    res_folder = load(self, "proto_root")
```

File: recipes/opentelemetry-proto/all/test_package/conanfile.py [hunk 3]
```diff
@@ -1,34 +1,24 @@
...
     def generate(self):
-        tc = CMakeToolchain(self)
-        tc.generate()
-        tc = CMakeDeps(self)
-        tc.generate()
-        tc = VirtualBuildEnv(self)
-        tc.generate(scope="build")
-
-        self._res_folder = self.dependencies["opentelemetry-proto"].conf_info.get("user.opentelemetry-proto:proto_root")
+        save(self, "proto_root", self.dependencies["opentelemetry-proto"].conf_info.get("user.opentelemetry-proto:proto_root"))
```
Comment (uilianries) [inline recipes/opentelemetry-proto/all/test_package/conanfile.py:19]:
```suggestion
        save(self, os.path.join(self.build_folder, "proto_root"), self.dependencies["opentelemetry-proto"].conf_info.get("user.opentelemetry-proto:proto_root"))
```

File: recipes/opentelemetry-proto/all/test_package/conanfile.py [hunk 4]
```diff
@@ -1,34 +1,24 @@
...
 
     def test(self):
-        # TODO: to remove in conan v2
-        if self._res_folder == "":
-            self._res_folder = self.deps_user_info["opentelemetry-proto"].proto_root
-
-        assert os.path.isfile(os.path.join(self._res_folder, "opentelemetry", "proto", "common", "v1", "common.proto"))
+        res_folder = load(self, "proto_root")
```
Comment (uilianries) [inline recipes/opentelemetry-proto/all/test_package/conanfile.py:22]:
```suggestion
        res_folder = load(self, os.path.join(self.build_folder, "proto_root"))
```

**Review by uilianries — COMMENTED** [review]
relative paths is mocking the test execution

File: recipes/opentelemetry-proto/all/conanfile.py [hunk 5]
```diff
@@ -1,43 +1,44 @@
...
 class OpenTelemetryProtoConan(ConanFile):
     name = "opentelemetry-proto"
     license = "Apache-2.0"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/open-telemetry/opentelemetry-proto"
     description = "Protobuf definitions for the OpenTelemetry protocol (OTLP)"
-    topics = ("opentelemetry", "telemetry", "otlp")
+    topics = ("opentelemetry", "telemetry", "otlp", "pre-built")
+
+    package_type = "application"
```
Comment (uilianries) [inline recipes/opentelemetry-proto/all/conanfile.py:18]:
```suggestion
    package_type = "build-scripts"
```

https://docs.conan.io/2/reference/conanfile/attributes.html#package-type

As this recipe is packaging only raw file, script fits better due the trait `visible=False`.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by toge — APPROVED** [review]

**Pull request merged**